### DISCUSSION TITLE
Small pull request for CloudReady

### DISCUSSION THREAD

**Discussion by Alex313031** [skycocker/chromebrew#4997] [Ideas]
Many people, including me, were on the forums about getting this to work with Cloudready. I also found it useful for dev builds of chromium os. I made a fork and just put a cloudready.md with info and links for what to do before running crew's install script. Recommending you add this to simplify things for new users. Add whatever you think is necessary I just made a simple .md

**Comment by cstrouse** [comment]
I don't know if we want to merge that since we don't support using crew on cloudready.

**Comment by satmandu** [comment]
Since Neverware is now part of Google, I think we SHOULD support Cloudready. Also, at some point the Apple Arm devices are going to make incredible cloudready devices.

**Reply by cstrouse** [reply]
The question is was it another anti-competitive move or are they really embracing an alternative build?

**Reply by Alex313031** [reply]
I believe anti-competitive. I hate ChromeOS Flex and it runs on none of my devices that Cloudready works fine on. Plus I liked the neverware logo and chromium logo

**Comment by satmandu** [comment]
Having said that, I'd like to avoid supporting Chromebrew on rootfs verification disabled devices. Not messing up the boot process or core ChromeOS filesystem dramatically reduces the support burden. Is there no way to enable a developer mode in Cloudready otherwise in Cloudready?

(I'm in dev mode on my ChromeOS device and I've kept rootfs verification enabled.)

**Reply by uberhacker** [reply]
I'm in agreement with @satmandu.  Disabling rootfs verification is the equivalent of tearing the seal and voiding the warranty.

**Comment by satmandu** [comment]
Also I don't know why I don's see the relevant PR here, but here's the doc you're talking about, right?

https://github.com/Alex313031/chromebrew/blob/master/cloudready.md

**Reply by uberhacker** [reply]
This definitely puts you in YMMV territory and support will be very difficult at this point if something is hosed.

**Reply by Alex313031** [reply]
Yes, that's what I'm talking about @satmandu . Also as @supechicken pointed out, chromium os and cloudready are already in "dev mode" so will edit the md, IDK what I was thinking. All this being said (in this whole thread) I was not able to get crew to work without disabling verity. The integration and execution of FS verity is slightly different on cloudready (different partition scheme, different noexec mounts, and custom, more UTD kernel). I don't know exactly why this would prevent crew from running properly (script would install crew sucessfully, but applications failed to install, whether precomp or from source) Tried on many machines with cloudready 78 thru latest @ 87.3. It works fine on chromium os and chrome os.  All this said, this IS YMMV, however I think it might be appropriate to at least put some info, maybe a link to the cros source md and Cloudready support page regarding FS verity, somewhere in crew's documentation.

**Reply by Alex313031** [reply]
@satmandu will add an addendum about the fact that it will most likely be unnecessary on baseline chromium os, but still keep the link to the info.

**Reply by satmandu** [reply]
It would be nice to figure out exactly what is breaking with cloudready & Chromebrew without enabling r/w rootfs.

Can you post some logs of the errors?

**Reply by cstrouse** [reply]
I think the primary issue was because they ran in guest mode 100% of the time. They now require authenticating with a Google account once before the system can be used in the usual fashion; not sure if this makes the previous issues moot but it's certainly possible. There were also some issues IIRC with symlinks or paths they modified that gave a bunch of trouble for some of the packages.

**Comment by supechicken** [comment]
Chromium OS devices (CloudReady also) are in dev mode already(unless you delete the `cros_debug` arg in GRUB boot options), we can access `crosh` shell without remounting rootfs as r/w.

**Reply by Alex313031** [reply]
Your'e right, I should correct this.
